stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -546,30 +551,52 @@ protected String pendingToString() {
       return listener;
     }
 
-    private Object waitForNext() throws InterruptedException {
+    private Object waitForNext() {
       if (threadless == null) {
```

Comment:
You could move the `if` statement inside the try-catch so it just wraps the `while` statement.